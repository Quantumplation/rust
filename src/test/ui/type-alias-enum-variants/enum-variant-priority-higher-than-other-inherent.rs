// Check that an `enum` variant is resolved, in the value namespace,
// with higher priority than other inherent items when there is a conflict.

enum E {
    V(u8)
}

impl E {
    fn V() {}
}

enum E2 {
    V,
}

impl E2 {
    const V: u8 = 0;
}

fn main() {
    <E>::V(); //~ ERROR arguments to this function are incorrect
    let _: u8 = <E2>::V; //~ ERROR mismatched types
}
