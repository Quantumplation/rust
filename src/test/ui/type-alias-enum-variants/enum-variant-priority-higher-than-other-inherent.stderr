error[E0308]: arguments to this function are incorrect
  --> $DIR/enum-variant-priority-higher-than-other-inherent.rs:21:5
   |
LL |     <E>::V();
   |     ^^^^^^^^ an argument of type u8 is missing
   |
note: tuple variant defined here
  --> $DIR/enum-variant-priority-higher-than-other-inherent.rs:5:5
   |
LL |     V(u8)
   |     ^^^^^
help: provide the argument
   |
LL |     V(u8)({u8});
   |     ^^^^^^^^^^^

error[E0308]: mismatched types
  --> $DIR/enum-variant-priority-higher-than-other-inherent.rs:22:17
   |
LL |     let _: u8 = <E2>::V;
   |            --   ^^^^^^^ expected `u8`, found enum `E2`
   |            |
   |            expected due to this

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0308`.
